content-services/content-sources-backend
# Fixes 4054: privatize repos without trailing slash

## Summary

This marks all public repos that are missing a trailing slash as not public.  This should cause them to be deleted if there are no repo configs pointing to them.  

## Testing steps

Create a duplicate:  
`make repos-import`
`make db-cli-connect`

```
insert into repositories (uuid, url, public) values (gen_random_uuid(), 'https://cdn.redhat.com/content/dist/rhel8/8.8/x86_64/appstream/os', true);
```

Now run the up migration.  It should no longer be public:

```
select count(*) from repositories where url = 'https://cdn.redhat.com/content/dist/rhel8/8.8/x86_64/appstream/os'
 and public = true;

```

0

```
select count(*) from repositories where url = 'https://cdn.redhat.com/content/dist/rhel8/8.8/x86_64/appstream/os'
 and public = false;
```
1


## Checklist
